cloud/scope/powervs_cluster_test.go

```diff
@@ -1093,6 +1095,395 @@ func TestGetResourceGroupID(t *testing.T) {
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When IBMVPCClient client CreateLoadBalancer returns error due to failed loadBalancer creation in cloud", func(*testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterAPIServerPort := int32(9090)
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient: mockVpc,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+
+					ResourceGroup: &infrav1beta2.IBMPowerVSResourceReference{
+						ID: core.StringPtr("test-resource-gid"),
+					},
+					LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
+						{
+							Name: "test-lb",
+							ID:   nil,
+						},
+					},
+					VPCSubnets: []infrav1beta2.Subnet{
+						{
+							Name: core.StringPtr("test-subnet"),
+							ID:   core.StringPtr("test-subnetid"),
+						},
+					},
+				},
+			},
+			Cluster: &capiv1beta1.Cluster{
+				Spec: capiv1beta1.ClusterSpec{
+					ClusterNetwork: &capiv1beta1.ClusterNetwork{
+						APIServerPort: &clusterAPIServerPort,
+					},
+				},
+			},
+		}
+
+		mockVpc.EXPECT().GetLoadBalancerByName(gomock.Any()).Return(nil, nil)
+		mockVpc.EXPECT().CreateLoadBalancer(gomock.Any()).Return(nil, nil, errors.New("Failed loadBalancer creation"))
+
+		requeue, err := clusterScope.ReconcileLoadBalancers()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When IBMVPCClient client CreateLoadBalancer successfully creates loadbalancer in cloud", func(*testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterAPIServerPort := int32(9090)
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient: mockVpc,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+
+					ResourceGroup: &infrav1beta2.IBMPowerVSResourceReference{
+						ID: core.StringPtr("test-resource-gid"),
+					},
+					LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
+						{
+							Name: "test-lb",
+							ID:   nil,
+						},
+					},
+					VPCSubnets: []infrav1beta2.Subnet{
+						{
+							Name: core.StringPtr("test-subnet"),
+							ID:   core.StringPtr("test-subnetid"),
+						},
+					},
+				},
+			},
+			Cluster: &capiv1beta1.Cluster{
+				Spec: capiv1beta1.ClusterSpec{
+					ClusterNetwork: &capiv1beta1.ClusterNetwork{
+						APIServerPort: &clusterAPIServerPort,
+					},
+				},
+			},
+		}
+
+		mockVpc.EXPECT().GetLoadBalancerByName(gomock.Any()).Return(nil, nil)
+		mockVpc.EXPECT().CreateLoadBalancer(gomock.Any()).Return(&vpcv1.LoadBalancer{
+			ID:                 core.StringPtr("test-lb-id"),
+			ProvisioningStatus: core.StringPtr("active"),
+		}, nil, nil)
+
+		requeue, err := clusterScope.ReconcileLoadBalancers()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).To(BeNil())
+
+		loadBalancer, ok := clusterScope.IBMPowerVSCluster.Status.LoadBalancers["test-lb"]
+		g.Expect(ok).To(BeTrue())
+		g.Expect(loadBalancer.State).To(BeEquivalentTo(infrav1beta2.VPCLoadBalancerStateActive))
```

Comment:
How about we verify the values of `ControllerCreated` and `Hostname` in status when the resource is created via controller?

Reply:
Done